oauth2-server/src/main/java/com/clouway/oauth2/codechallenge/AuthorizationCodeVerifier.java

```diff
@@ -0,0 +1,53 @@
+package com.clouway.oauth2.codechallenge;
+
+import com.google.common.io.BaseEncoding;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * @author Vasil Mitov <vasil.mitov@clouway.com>
+ */
+public class AuthorizationCodeVerifier implements CodeVerifier {
+  @Override
+  public boolean verify(CodeChallenge codeChallenge, String codeVerifier) {
+    try {
+      //no code verifier was provided and no code challenge was saved so this is a normal OAuth2 code flow without PKCE
+      if (isNullOrEmpty(codeVerifier) && isNullOrEmpty(codeChallenge)) {
+        return true;
+      }
+
+      if (isNullOrEmpty(codeVerifier) && !isNullOrEmpty(codeChallenge)) {
+        //there is a saved code challenge but provided codeVerifier is null or empty.
```

Comment:
comment should be moved before the if statement as in previous block

Reply:
Moved.